guides/deployment/fly.md

```diff
@@ -0,0 +1,281 @@
+# Deploying on Fly
+
+## What we'll need
+
+The only thing we'll need for this guide is a working Phoenix application. For those of us who need a simple application to deploy, please follow the [Up and Running guide](up_and_running.html).
+
+## Goals
+
+Our main goal for this guide is to get a Phoenix application running on Fly.io.
+
+## Steps
+
+Let's separate this process into a few steps so we can keep track of where we are.
+
+- Install the Fly CLI
+- Sign up for Fly
+- Make our project ready for Fly
+- Create and set up our Fly application
+- Provision a database
+- Deploy time!
+- Helpful Fly resources
+
+## Installing the Fly CLI
+
+Follow the instructions [here](https://fly.io/docs/getting-started/installing-flyctl/) to install the command-line interface for the Fly platform.
+
+## Sign up for Fly
+
+We can [sign up for an account](https://fly.io/docs/getting-started/login-to-fly/) using the CLI.
+
+```console
+$ fly auth signup
+```
+
+Fly has a [free tier](https://fly.io/docs/about/pricing/) for applications without a database. A credit card is required when setting up an account to help prevent abuse. See the [pricing](https://fly.io/docs/about/pricing/) page for more details.
+
+## Make our project ready for Fly
+
+For this guide, we'll use a Dockerfile and build a release for our Fly deployment. Internally, Fly's networking uses IPv6, so there is a little config we can do to our application to make it a smooth experience.
```

Comment:
Keep in mind that the inet6 config has been added to Phoenix master generators, so everything should be inet6 ready by the time this guide is up. :)

Reply:
@josevalim Regarding the master branch docs and legacy configuration, is it preferred to _only_ document according to the _current_ approach or should there be explanation regarding previous versions? I get the sense that you lean towards only document based on the current state.

Reply:
The previous versions will still work with the new config. So I would move to the new config because that’s the best practice and then leave the important notes on the important places (ipv6 and inet6)

Reply:
I updated the `config/runtime.exs` example to be based off the current version is master.